Objects/cellobject.c

```diff
@@ -140,8 +140,19 @@ cell_get_contents(PyCellObject *op, void *closure)
     return op->ob_ref;
 }
 
+int
+cell_set_contents(PyCellObject *op, PyCellObject *obj)
+{
+    Py_INCREF(obj);
+    PyCellObject *tmp_op = (PyCellObject *)(op->ob_ref);
```

Comment:
You may have used the macro incorrectly. Did you try `Py_SETREF(op->ob_ref, obj)`?
Actually, you need `Py_XSETREF` (and `Py_XINCREF`), since `obj` can be NULL as @serhiy-storchaka pointed out.